presto-hive/src/main/java/io/prestosql/plugin/hive/util/HiveUtil.java

```diff
@@ -670,16 +672,18 @@ public static ConnectorViewDefinition decodeViewData(String data)
 
     public static ConnectorViewDefinition buildHiveViewConnectorDefinition(HiveCatalogName catalogName, Table view)
     {
+        String viewText = view.getViewExpandedText()
+                .orElseThrow(() -> new PrestoException(HIVE_INVALID_METADATA, "No view expanded text: " + view.getSchemaTableName()));
         return new ConnectorViewDefinition(
-                translateHiveQlToPrestoSql(view.getViewExpandedText().get()),
+                translateHiveQlToPrestoSql(viewText),
                 Optional.of(catalogName.toString()),
                 Optional.ofNullable(view.getDatabaseName()),
                 view.getDataColumns().stream()
-                        .map(column -> new ConnectorViewDefinition.ViewColumn(column.getName(), TypeId.of(column.getType().getTypeSignature().toString())))
+                        .map(column -> new ViewColumn(column.getName(), TypeId.of(column.getType().getTypeSignature().toString())))
                         .collect(toImmutableList()),
-                Optional.ofNullable(view.getOwner()),
-                Optional.empty(), // no comment
-                false); // don't run as invoker);
+                Optional.ofNullable(view.getParameters().get(TABLE_COMMENT)),
+                Optional.of(view.getOwner()),
```

Comment:
@sopel39 comment/owner were reversed. Should  we have a test? https://github.com/prestosql/presto/issues/2753

Reply:
good idea